src/main/scala/codacy/pylint/Pylint.scala

```diff
@@ -10,50 +11,142 @@ import scala.util.{Properties, Success, Try}
 
 object Pylint extends Tool {
 
-  override def apply(path: Path, conf: Option[Seq[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[Iterable[Result]] = {
-    def isEnabled(issue: Issue) =
-      conf.map(_.exists(_.patternId == issue.patternId)).getOrElse(true)
+   override def apply(path: Path, conf: Option[Seq[PatternDef]], files: Option[Set[Path]])(implicit spec: Spec): Try[Iterable[Result]] = {
+    def isEnabled(issue: Result) = {
+       issue match {
+          case Issue(_, _, patternId, _) => conf.map(item => item.exists(_.patternId == patternId)).getOrElse(true)
+          case _ => true
+       }
+    }
 
-    commandFor(path, conf, files).flatMap { case cmd =>
-      val wrappedCmd = Seq("bash", "-c", cmd.mkString(" "))
+    def buildFileCommands(files: Map[String, Array[String]]) = {
+      files.map { case (key, values) => commandFor(key, path, conf, values)}
+           .flatMap( item => item.toOption)
+    }
 
-      Try(wrappedCmd.lineStream_!(ProcessLogger(_ => ()))).map { lines =>
-        lines.flatMap(parseLine).filter(isEnabled)
+    def getStdout(command: Seq[String]): Try[Seq[String]] = {
+      Try {
+          CommandRunner.exec(command) match {
+            case Right(resultFromTool) =>
+              resultFromTool.stdout
+            case Left(failure) => {
+              throw failure}
+          }
       }
     }
+
+    val collectedFiles = collectFiles(files, path)
+    val classified = classifyFiles(collectedFiles)
+    val commands = classified.map { case item => buildFileCommands(item) }
+    val lines_iterable = commands.map { item => item.map(getStdout) }
+    val lines = lines_iterable.map {
+      case iterable => iterable.flatMap {
+          case item => item.toOption
+      }.flatten
+    }
+    lines.map { case line => line.flatMap(parseLine).flatten.filter(isEnabled)}
   }
 
-  private implicit lazy val writer = Json.reads[Issue]
 
+  private implicit lazy val writer = Json.reads[Issue]
+  
   private def parseLine(line: String) = {
-     val elements = line.split("###")
-     Try(Issue(SourcePath(elements(0)),
-               ResultMessage(elements(2)),
-               PatternId(elements(3)),
-               ResultLine(elements(1).toInt))).toOption
-
+     val LineRegex = """(.*?)###(.*?)###(.*?)###(.*?)""".r
+     line match {
+         case LineRegex(filename, lineNumber, message, patternId) if message.contains("invalid syntax") =>
+           val fileError = FileError(SourcePath(filename),
+                                     Option(ErrorMessage(message)))
+           val issue = Issue(SourcePath(filename),
+                              ResultMessage(message),
+                              PatternId(patternId),
+                              ResultLine(lineNumber.toInt))
+           Option(Seq(fileError, issue))
+         case LineRegex(filename, lineNumber, message, patternId) =>
+           Option(Seq(Issue(SourcePath(filename),
```

Comment:
Here, since you added my suggestion of using `Option` you do not need `Option(Seq(....))` use can just `Option(.....)`